tests/support/router.ts

```diff
@@ -14,8 +14,13 @@ export const expectBackAndForwardNavigationWorks = async (
   page: Page,
 ) => {
   const currentURL = page.url();
+
   await page.goBack();
+  // Wait for the loading bar to disappear.
+  await expect(page.locator(".loading-bar.visible")).toHaveCount(0);
```

Comment:
94c791c6b5c21ad6ff0dd019651fd0d5bd6818ea

> Finally, you’re not checking that the loading bar has been visible at some point. This may be hard to achieve though since the loading times are quite fast and you may miss it being visible. This is worth a comment in the code though.

We're actually in a test helper function here. The actual goal of this helper is to check that when going back-and-forth between routes the URL is always updated correctly. I had to add these extra `expect`s to make sure we check the URL only after loading has finished, otherwise we would get the old URL and the test would fail. I added a comment to better explain this. Let me know if it's clear enough.

Reply:
That makes sense. Maybe even more of a reason to use `waitFor()` instead of `expect()`.

Reply:
Right, missed that suggestion. This makes things even more clear, removed the comment. 62e47a5f77ecc30e43d6188e911ab030cf777147.